--- Modules/Compiler/Clang-CXX-CXXImportStd.cmake	2026-02-08 13:52:19.542007143 +0100
+++ Modules/Compiler/Clang-CXX-CXXImportStd.cmake.patched	2026-02-08 13:53:09.078767962 +0100
@@ -9,10 +9,12 @@
   endif ()
 
   if (NOT DEFINED CMAKE_CXX_STDLIB_MODULES_JSON)
+    separate_arguments(_cxx_flags_list NATIVE_COMMAND "${CMAKE_CXX_FLAGS}")
     execute_process(
       COMMAND
         "${CMAKE_CXX_COMPILER}"
         ${CMAKE_CXX_COMPILER_ID_ARG1}
+        ${_cxx_flags_list}
         "-print-file-name=${_clang_modules_json_impl}.modules.json"
       OUTPUT_VARIABLE _clang_libcxx_modules_json_file
       ERROR_VARIABLE _clang_libcxx_modules_json_file_err
